src/main/java/fact/io/hdureader/HDU.java

```diff
@@ -113,11 +118,7 @@ public BinTable getBinTable(){
      * @return the number of bytes to skip to hte next hdu.
      */
     long offsetToNextHDU(){
-        if(sizeOfDataArea() == 0){
-            return 0;
-        }
-        long numberOfBlocks = sizeOfDataArea() / 2880;
-        return 2880 * numberOfBlocks + 2880;
+        return ((sizeOfDataArea()+(FITS_BLOCK_SIZE-1))/FITS_BLOCK_SIZE)*FITS_BLOCK_SIZE;
```

Comment:
isn't this just the same as: sizeOfDataArea()+FITS_BLOCK_SIZE-1)
Or are there numerical reasons?

Reply:
numerical, all are ints

Reply:
Yes, you are correct. The `/ FITS_BLOCK_SIZE)*FITS_BLOCK_SIZE` should not be necessary and just add noise

Reply:
> numerical, all are ints

Can you explain? Why do you need to add not needed multiplications / divisions because they are ints?

Reply:
It's totally not obvious from the code. Add least split it to multiple lines and add comments to make clear what this is supposed to do.

Reply:
its a widely used trick to calculate the ceil(a,b) of two ints faster, it is just that nowadays you rarely see it done, so people don't know about it as much, i added a comment explaining it a little bit